docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+**Developer Guide**
```

Comment:
The DG is well written with breakdown on architecture and implementation. I think the group can focus on improving the format of the information written. There seems to be an issue on the images as they were not rendered properly so I could not give comment on the use of visuals. Keep up the good work! Good Job :))